### DISCUSSION TITLE
Virtual Machine Low Disk Space

### DISCUSSION THREAD

**Discussion by LucasTheIntern** [bitcraze/discussions#1085] [Q&A] [answered]
Hello, 

I'm using the latest version of the VM, and running into an issue where I cannot install things due to low remaining disk space. When I boot up the VM, I get the following message as a pop-up:
![popup](https://github.com/bitcraze/discussions/assets/120214901/33994df2-0659-405b-9a29-328d540ea996)

I'm trying to complete the instructions on the "getting started with the AI deck" tutorial as a precursor to the CPX communication example tutorial:
https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/
All went well until I got to the section titled "Setting up docker and the autotiler", where running the command: 
`$ docker run --rm -it --name myAiDeckContainer bitcraze/aideck`
yields the result:
![terminaloutput](https://github.com/bitcraze/discussions/assets/120214901/8aa29031-ff17-4d3b-805b-739b43536fa4)
Which makes sense, based on the popup.

I proceeded to tackle the issue by shutting down the VM and attempting to increase the allocated storage from the virtualbox manager:
![vboxmanager](https://github.com/bitcraze/discussions/assets/120214901/b9bccba3-0de8-49b8-adf3-6ad3da1a134f)
I was able to adjust the slider and decided give it 75 GB (later tried increasing it again to 77 GB). Sadly, the popup returned when I booted the VM up again, and I got the same error from the terminal. A labmate suggested that I try installing gparted and following this tutorial:
https://www.baeldung.com/linux/virtualbox-resize-hard-disk

I did as suggested, but the issue remains unresolved. I was able to resize /dev/sda3 to include the new storage (~50 GB?), but the machine still doesn't seem to be using it. After adding the extra 2 GB, it appears under "unallocated" as shown here:
![gparted](https://github.com/bitcraze/discussions/assets/120214901/bf72b43d-6a42-4ba4-8770-62e328f602f6)

I had the idea that perhaps I need to be resizing the space named "/dev/sda2", but am unable to do so - the unallocated space is not available in the resizing menu:
![resizingmenu](https://github.com/bitcraze/discussions/assets/120214901/40fdbf00-19d4-4f9b-b192-dbb023bee595)
But it is available in the same menu for /dev/sda3.

If anyone has had this issue and can instruct me on how to rectify it, your help would be very much appreciated!

**Comment by knmcguire** [comment, marked as answer]
Hmm yeah, this seems to be a bug. The VM should have way more space available for the file system but for some reason, this was not added at the beginning. I made an issue about it here: https://github.com/bitcraze/bitcraze-vm/issues/67

Basically, you've got to do the following in a terminal in the VM:
```
sudo lvextend -l +100%FREE /dev/ubuntu-vg/ubuntu-lv
sudo resize2fs /dev/ubuntu-vg/ubuntu-lv
```

And this needs to be redone every time you add disk space to the virtualbox vm and add it to the partition with gparted.

I'm currently trying out downloading the dockerimage again to check this fix out :)

**Reply by knmcguire** [reply]
credit to @ataffanel as he figured out the fix
